R/build_collection.R

```diff
@@ -1,27 +1,165 @@
 
 
-build_collection <- function(title,
-                             links = build_links(),
-                             assets = NULL,
-                             extent) {
+build_collection <- function(id,
+                             links = NULL,
+                             title,
+                             assets = build_assets(tnail_object, tnail_title, parquet_uri, p_title, p_description),
+                             extent = NULL,
+                             keywords = NULL,
+                             providers = NULL,
+                             summaries = NULL,
+                             description = NULL,
+                             item_assets = NULL,
+                             table_columns,
+                             #extensions = NULL,
+                             publications = NULL) {
 
-  list(id = id,
-       type = "Collection",
-       title = title,
-       assets = assets,  #REMEMBER TO COLLAPSE NULLS
-       )
+  collection_list <- list(id = id,
+                          type = "Collection",
+                          links = NULL,
+                          title = title,
+                          assets = assets, #REMEMBER TO COLLAPSE NULLS
+                          extent = extent,
+                          license = "proprietary",
+                          keywords = keywords,
+                          providers = providers,
+                          summaries = summaries,
+                          description = description,
+                          item_assets = item_assets,
+                          stac_version = '1.0.0',
+                          table_columns = tables, ##need to use table:columns but syntax confused R...fix later
+                          group_id = 'FLARE_forecast_id', ##just a placeholder for now
+                          stac_extensions = list('0' = "https://stac-extensions.github.io/scientific/v1.0.0/schema.json",
+                                                 '1' = "https://stac-extensions.github.io/item-assets/v1.0.0/schema.json",
+                                                 '2' = 	"https://stac-extensions.github.io/table/v1.2.0/schema.json"),
+                          publications = publications
+  )
+
+  ## WRITE COLLECTION LIST INTO JSON FORMAT
```

Comment:
It's probably okay to let all the constructors just retain list objects.  think of writing into JSON as the same as writing to CSV -- inside R we don't want "csv"-type objects, we want `data.frame` / `tibble` objects.  JSON is just the natural text-based serialization of a "list" in R in the same way that csv is the natural/simplest serialization of a 'data.frame'